models/iosxr/ntp_global/iosxr_ntp_global.yaml

```diff
@@ -0,0 +1,230 @@
+---
+module: iosxr_ntp_global
+short_description: Manages ntp resource module
+description: This module configures and manages the attributes of  ntp on Cisco
+  IOSXR platforms.
+version_added: 2.5.0
+author: Ashwini Mhatre (@amhatre)
+notes:
+  - Tested against IOSXR 7.0.2.
+  - This module works with connection C(network_cli).
+options:
+  config:
+    description: A dictionary of ntp options
+    type: dict
+    suboptions:
+        access_group:
+          description: Control NTP access
+          type: list
+          elements: dict
+          suboptions:
+            ipv4: &ipv4
+              type: dict
+              description: Configure IPv4 access
+              suboptions:
+                peer: &peer
+                  type: str
+                  description: Provide full access
+                query_only: &query_only
+                  type: str
+                  description: Allow only control queries.
+                serve: &serve
+                  type: str
+                  description: Provide server and query access.
+                serve_only: &serve_only
+                  type: str
+                  description: Provide only server access.
+            ipv6: &ipv6
+              type: dict
+              description: Configure IPv6 access
+              suboptions:
+                peer: *peer
+                query_only: *query_only
+                serve: *serve
+                serve_only: *serve_only
+            vrf:
+              type: dict
+              description: Specify non-default VRF.
+              suboptions:
+                name:
+                  type: str
+                  description: Specify non-default VRF.
+                ipv4: *ipv4
+                ipv6: *ipv6
+        authenticate:
+          description: Authenticate time sources
+          type: bool
+        authentication_key:
+          description: Authentication key for trusted time sources
+          type: dict
+          suboptions:
+            id:
+              description: <1-65535>  Key number
+              type: int
+            key:
+              description: Authentication key.
+              type: str
+              no_log: True
```

Comment:
*Note*: This will trigger validate-modules since no_log is not a part of doc schema (yet). You will have to remove this from the docstring once scaffolding is done.